### PR TITLE
Make Basic Usage tutorial more detail.

### PR DESCRIPTION
**Pull request by jenhaoyang**
Use the Hello example as the example of Basic Usage. It might be helpful for Python user who have limited C++ experience.

### CODE REVIEW

**Review by thewtex — APPROVED** [review]
@jenhaoyang thank you for this contribution!

File: docs/usage.rst [hunk 1]
```diff
@@ -20,19 +20,31 @@ Basic Usage
 
 Example of setup.py, CMakeLists.txt and pyproject.toml
 ------------------------------------------------------
+The full example code is `Here <https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/hello-cpp>`_
 
-To use scikit-build in a project, place the following in your project's
+Make a fold name my_project as your project root folder, place the following in your project's
```
Comment (jcfr) [inline docs/usage.rst:25]:
```suggestion
Create a directory named ``my_project`` as your project root folder, then add the following in your project's
```

File: docs/usage.rst [hunk 2]
```diff
@@ -20,19 +20,31 @@ Basic Usage
...
+        python_requires=">=3.7",
+    )
 
 Your project now uses scikit-build instead of setuptools.
 
 Next, add a ``CMakeLists.txt`` to describe how to build your extension. In the following example,
 a C++ extension named ``_hello`` is built::
 
-    cmake_minimum_required(VERSION 3.11.0)
+    cmake_minimum_required(VERSION 3.4...3.22)
```
Comment (jcfr) [inline docs/usage.rst:44]: This contradicts the requirement specified below `cmake>=3.18`

Comment (jcfr) [inline docs/usage.rst:44]:
```suggestion
    cmake_minimum_required(VERSION 3.18.6)
```

File: docs/usage.rst [hunk 3]
```diff
@@ -43,13 +55,32 @@ Then, add a ``pyproject.toml`` to list the build system requirements::
...
     [build-system]
     requires = [
-      "setuptools>=42",
-      "scikit-build",
-      "cmake",
-      "ninja; platform_system!='Windows'"
+        "setuptools>=42",
+        "scikit-build>=0.13",
+        "cmake>=3.18",
+        "ninja",
```
Comment (jcfr) [inline docs/usage.rst:61]:
```suggestion
        "ninja; platform_system!='Windows'",
```

This contradicts the comment below and should be maintained.

File: docs/usage.rst [hunk 4]
```diff
@@ -43,13 +55,32 @@ Then, add a ``pyproject.toml`` to list the build system requirements::
...
     ]
     build-backend = "setuptools.build_meta"
 
+Make a hello folder inside my_project folder and place `_hello.cxx <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/_hello.cxx>`_ and `__init__.py <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/__init__.py>`_ inside hello folder.
```
Comment (jcfr) [inline docs/usage.rst:65]:
```suggestion
Create a ``hello`` directory inside ``my_project`` root directory folder and place the files `_hello.cxx <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/_hello.cxx>`_ and `__init__.py <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/__init__.py>`_ inside the ``hello`` directory.
```

File: docs/usage.rst [hunk 5]
```diff
@@ -43,13 +55,32 @@ Then, add a ``pyproject.toml`` to list the build system requirements::
...
+
+Now every thing is ready, go to my_project's parent folder and type following command to install your extension::
```
Comment (jcfr) [inline docs/usage.rst:67]:
```suggestion
Now every thing is ready, go to my_project's parent directory and type the following command to install your extension::
```

**Pull request merged**